### PR TITLE
feat(save): remove dead code to prevent nil value on save

### PR DESCRIPTION
**Pull request by asbjornhaland**
`get_config()` is removed and replaced by
`get_term_config()` and `get_mark_config()`

### CODE REVIEW

File: lua/harpoon/init.lua [hunk 1]
```diff
@@ -43,15 +43,15 @@ function merge_table_impl(t1, t2)
     end
 end
 
-function merge_tables(...)
```
Comment (ThePrimeagen) [inline lua/harpoon/init.lua:46]: Is my remove whitespace not being called?  Double you Tee Eff!!! let me fix that.

**Comment by ThePrimeagen** [comment]
and rubber stamp

**Comment by ThePrimeagen** [comment]
And actually rubber stamp

File: lua/harpoon/init.lua (hunk 1 above)
Comment (asbjornhaland) [reply lua/harpoon/init.lua:46]: Seems like it. I did remove in several files.

**Pull request merged**